mobly/base_suite.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+
+class BaseSuite(abc.ABC):
```

Comment:
this is adding a new layer of abstraction between the test classes and the suite runner right?

We should promote the generated suite way as the default option to reduce boilerplate code for suites. So this class should only be used when configs need to be modified between classes in some way.

All these context should be clearly stated (maybe even repeatedly) to avoid confusion.

Do we need to introduce the generated suite feature in the suite runner?